src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php

```diff
@@ -29,10 +29,16 @@ interface DenormalizerInterface
     /**
      * Denormalizes data back into an object of the given class.
      *
-     * @param mixed       $data    Data to restore
-     * @param string      $type    The expected class to instantiate
-     * @param string|null $format  Format the given data was extracted from
-     * @param array       $context Options available to the denormalizer
+     * @template TObject of object
+     *
+     * @param mixed                             $data    Data to restore
+     * @param class-string<TObject>|string|null $type    The expected class to instantiate
```

Comment:
```suggestion
     * @param class-string<TObject>|string $type    The expected class to instantiate
```

As far as I see, this can't be null?